<a href="https://colab.research.google.com/github/dgomezbo/Diplomado_Ciencia_Datos/blob/main/Copia_de_dgomezgo_Tarea_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Tarea 2: Medidas estadísticas de dispersión: la desviación estándar**

![image.png](attachment:24864cbb-0353-44b5-9800-5963a77c9ff0.png)

Fuente: [Wikipedia] Google Images: https://www.google.com.co/url?sa=i&url=https%3A%2F%2Fes.wikipedia.org%2Fwiki%2FMedidas_de_dispersi%25C3%25B3n&psig=AOvVaw3eOjrnV0Yqr8rFEE7_ygII&ust=1691428868986000&source=images&cd=vfe&opi=89978449&ved=2ahUKEwix5brYxciAAxVsSTABHXzkAb8QjRx6BAgAEAw


</center>


*Autor: David Manuel Gómez Bolívar*

![image.png](attachment:33ca899e-8fa7-4cd0-8d2f-b03d89c602ef.png)


Experto financiero y gestor innovador, con comunicación eficaz y facilidad para solución de problemas, así como con alta capacidad de coordinación y planeación de equipos de trabajo. Autodidacta más allá de múltiples diplomados y cursos en gerencia, finanzas, métodos cuantitativos y tecnología. Fortaleza en Finanzas Corporativas, Mercado de Dinero y de Capitales, Gestión de Riesgo, Project Finance, US GAAP / IFRS y promotor de mejores prácticas internacionales en Estructuración Financiera, Métodos Cuantitativos, Derivados Financieros, Opciones Reales, Asset Management, Valoración de Empresas y de Activos Financieros, e Inteligencia Artificial.

Trabajador en equipo, con liderazgo, adaptable a nuevos ambientes, aprendizaje continuo, y con alta habilidad para transferencia de conocimiento. Experiencia en diferentes sectores (financiero, real, consultoría, auditoría y público).

Experiencia como docente universitario de Mercados Financieros y de Capitales, Financiación, Decisiones de Inversión, Opciones Reales, Riesgo, Evaluación de Proyectos de Inversión, Estrategia Financiera, Organizaciones Económicas Internacionales, Bloomberg, entre otros; así como en capacitaciones sobre IFRS, Derivados Financieros, Finanzas Corporativas y Startups. Certificado en Bloomberg Markets Concepts (BMC).



*Desviación estándar*

Si bien es cierto las medidas de tendencia central son las más conocidas por la mayoría de "ciudadanos a pie", es primordial entender la desviación estándar como una de las medidas estadísticas de otra familia, la cual se denomina **dispersión**.

A partir de ello, la desviación estándar refleja el nivel de cercanía (o separación) que tienen los datos (variables) observados con respecto a su media aritmética (promedio).

De esta forma, en dos muestras que comparten un promedio igual, se pueden observar dos desviaciones estándar diferentes. Esto porque quizás una muestra tiene menor nivel de *separación* entre sus variables, con respecto a su promedio; que la otra muestra. Esto es relevante a nivel de finanzas, dado que la desviación estándar es la primera medida estadística que infiere un concepto muy importante para el riesgo: **volatilidad**.

Sin embargo, se debe tener en cuenta que la medición estadística de dispersión primaria es la varianza, de la cual, al calcular su raíz cuadrada, se obtiene la desviación estándar. A continuación, su fórmula:

![image.png](attachment:ee802162-325e-4eda-affe-05f22a120764.png)

![image.png](attachment:d9ea112c-d85b-452a-8bf7-a4d25ff879f0.png)

Fuente: Economipedia, en https://www.google.com.co/imgres?imgurl=https%3A%2F%2Feconomipedia.com%2Fwp-content%2Fuploads%2FMedidas-de-dispersi%25C3%25B3n-estad%25C3%25ADstica.png&tbnid=JSiEMP6gBtJYaM&vet=12ahUKEwiiuPjSxciAAxWrjrAFHcOGCVwQMygAegUIARDNAQ..i&imgrefurl=https%3A%2F%2Feconomipedia.com%2Fdefiniciones%2Fmedidas-de-dispersion.html&docid=delTWnY-7He9-M&w=1080&h=1080&q=medida%20estad%C3%ADstica%20de%20dispersi%C3%B3n&hl=es&authuser=0&ved=2ahUKEwiiuPjSxciAAxWrjrAFHcOGCVwQMygAegUIARDNAQ

En dicho contexto, la desviación estándar tiene la siguiente fórmula:

![image.png](attachment:484d531b-1d09-499f-ab7c-848e89856c52.png)

![image.png](attachment:0f571d32-b5b9-4a34-8ff5-61575b58e8e9.png)

Fuente: Economipedia, en https://www.google.com.co/imgres?imgurl=https%3A%2F%2Feconomipedia.com%2Fwp-content%2Fuploads%2FMedidas-de-dispersi%25C3%25B3n-estad%25C3%25ADstica.png&tbnid=JSiEMP6gBtJYaM&vet=12ahUKEwiiuPjSxciAAxWrjrAFHcOGCVwQMygAegUIARDNAQ..i&imgrefurl=https%3A%2F%2Feconomipedia.com%2Fdefiniciones%2Fmedidas-de-dispersion.html&docid=delTWnY-7He9-M&w=1080&h=1080&q=medida%20estad%C3%ADstica%20de%20dispersi%C3%B3n&hl=es&authuser=0&ved=2ahUKEwiiuPjSxciAAxWrjrAFHcOGCVwQMygAegUIARDNAQ

**Ejemplo**

Después de importar los módulos *Pandas* y *Openpyxl*, importaremos un archivo muy sencillo de Excel, para poder calcular la media aritmética y la desviación estándar.

In [ ]:
import pandas as pd

In [ ]:
pip install openpyxl

In [ ]:
import openpyxl

In [ ]:
#ejemplo = pd.read_excel(io=r'C:\Personal PC\Posgrado\UNAL 2023\Tareas\dgomezbo Ejemplo Tarea2.xlsx', sheet_name='Base')
ejemplo = pd.read_excel(io=r'/dgomezbo Ejemplo Tarea2.xlsx', sheet_name='Base')

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
print(ejemplo)

   Muestra A  Muestra B
0          1          0
1          2          1
2          3          3
3          5          5
4          6          6
5          7          7
6          8          8
7         12         12
8         14         15
9         15         16


In [ ]:
ejemplo.dtypes #tipo de objeto en cada columna
print("\n")
print(ejemplo.describe())
print("\n")



       Muestra A  Muestra B
count  10.000000  10.000000
mean    7.300000   7.300000
std     4.945256   5.538752
min     1.000000   0.000000
25%     3.500000   3.500000
50%     6.500000   6.500000
75%    11.000000  11.000000
max    15.000000  16.000000




Como se puede observar, las 2 muestras tienen la misma media aritmética (7.3), pero su desviación estándar es diferente (4.94 vs 5.54).

Muchas gracias.

Referencia de módulo openpyxl: *YouTube Channel - Grow up: Carga Datos de Excel a Python de diferentes formas*